ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js

```diff
@@ -356,16 +360,20 @@ module.exports = Marionette.LayoutView.extend({
                     }
                     break;
                 case 'INTERSECTS':
-                    var filterValue = typeof(filter.value) === 'string' ? filter.value : filter.value.value;
-                    locationModel = new LocationModel({
-                        polygon: CQLUtils.arrayFromPolygonWkt(filterValue),
-                        color: color
-                    });
-                    this.map.showPolygonShape(locationModel);
+                    this.handleFilterAsPolygon(filter.value, color, filter.distance);
                     break;
             }
         }
     },
+    handleFilterAsPolygon(value, color, distance) {
+        const filterValue = typeof(value) === 'string' ? value : value.value;
+        const locationModel = new LocationModel({
+            polygon: CQLUtils.arrayFromPolygonWkt(filterValue),
+            color: color,
+            ...(distance && {polygonBufferWidth: distance})
```

Comment:
❓ Do we need to do the check here?

Reply:
Are you asking whether we need this guard at all or whether it should go up a couple of lines into it's own block?